src/util/numeric.h

```diff
@@ -446,3 +448,18 @@ inline irr::video::SColor multiplyColorValue(const irr::video::SColor &color, fl
 			core::clamp<u32>(color.getGreen() * mod, 0, 255),
 			core::clamp<u32>(color.getBlue() * mod, 0, 255));
 }
+
+template <typename T> inline T numericAbsolute(T v) { return v < 0 ? T(-v) : v;                }
+template <typename T> inline T numericSign(T v)     { return T(v < 0 ? -1 : (v == 0 ? 0 : 1)); }
```

Comment:
`std::abs` and can probably do the latter with `std::copysign`

Reply:
neither of these are templates. std::abs is probably defined for enough types that we could probably get away with it (tho casting would be involved for anything smaller than an int), but numericSign is designed only for floats. i'm kind of opposed to needlessly casting numbers from integers to floats and then back again just to extract the sign

still, c++ is so picky about types (especially when it comes to resolving overloads) that i'm hesitant to use a non-template for these things

Reply:
e.g., you're trying to encode the sign of an `int8_t` into a packet, and do `pkt << std::copysign(myint)` on one end, then do `readI8(is)` on the other end, and then the deserializer breaks for no apparent reason until you figure out that `myint` was being silently converted to float, and thus the sign was encoded as a float in the packet because the compiler resolved a different `operator<<` than you intended, so `readI8()` was trying to decode part of a float as an unsigned byte and leaving the rest of the float in the stream for the next function to choke on.

similar issues could happen with `std::abs` for types smaller than int or for any(!) unsigned types. there's a reason a lot of people are wary of using the STL in complex projects, and this is exactly the sort of project where i worry this kind of stuff could cause very difficult-to-find problems down the road

Reply:
I doubt it'd cause concrete problems for *us* but those are some good reasons so it's fine.